search/api.py

```diff
@@ -129,3 +163,14 @@ def reformat(key, values, missing_count=None):
         ret_dict[key] = reformat((key, label), values)
 
     return ret_dict
+
+
+def get_vocab_name(vocab_id):
+    """Get vocabulary name by ID."""
+    vocab = Vocabulary.objects.get(id=vocab_id)
+    return vocab.name
+
+
+def get_term_label(term_id):
+    """Get term label by ID."""
+    return Term.objects.get(id=term_id).label
```

Comment:
The new code does store the full name in Elasticsearch. This is only because the Haystack queries use the primary keys. These will be deleted entirely along with the Haystack code.